### PR TITLE
Ensure coms-id tags buried in history are used when soft-deleted

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR focuses on making sure that when objects are soft deleted that their previous coms-id values are still used during synchronization where appropriate.

- 50f62e4 Implement _deriveObjectId utility synchronize method
- 58cfed0 Ensure that isAtPath utility recognizes files at root level as valid
- aff9799 Drop partial implementation of all queryparam from syncBucket controller

[SHOWCASE-3317](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-3317)

## Types of changes

Bug fix (non-breaking change which fixes an issue)
New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by norrisng-bc — APPROVED** [review]

File: app/src/services/sync.js [hunk 1]
```diff
@@ -81,7 +127,8 @@ const service = {
             response[0].versions.find(v => v.id === version.id).metadata = metadata;
           }
         }
-        log.verbose(`Finished syncing  ${path} in bucket ${bucketId}`, { function: 'syncJob', result: response });
+
+        log.verbose(`Finished syncing ${path} in bucket ${bucketId}`, { function: 'syncJob', result: response });
         return response;
```
Comment (TimCsaky) [inline app/src/services/sync.js:132]: we don't actually expose this response to the api user.. but it seems like a nice thing to have (a nested array of object>version?meta/tags) for logging.

File: app/src/services/sync.js [hunk 2]
```diff
@@ -91,95 +138,66 @@ const service = {
...
-   * @param {string} [options.bucketId] The uuid bucketId of object in sync job
+   * Synchronizes Object level data
+   * @param {string} options.path The path of object in sync job
+   * @param {string | null} options.bucketId The uuid bucketId of object in sync job
    * @param {string} [options.userId=SYSTEM_USER] The uuid of a user that created the sync job
    * @param {object} [etrx=undefined] An optional Objection Transaction object
-   * @returns {object} synced objects that exist in COMS and S3 eg:
-   * <ObjectModel> (synced object)
-   * or `undefined` (when object was pruned from COMS db after sync)
+   * @returns {Promise<Array<object | undefined>>} Either an array of synced objects or undefined
```
Comment (TimCsaky) [inline app/src/services/sync.js:148]: i think this syncObject method returns just one js object 
`@returns {Promise<object | undefined>}`

File: app/src/services/sync.js [hunk 3]
```diff
@@ -14,41 +14,87 @@ const versionService = require('./version');
...
+
+        // 1. Sync Object
+        const { newObject, ...object } = await service.syncObject({ path: path, bucketId: bucketId, userId: userId }, trx);
         response.push(object);
 
-        // 2. sync all versions of object
+        // 2. Sync Object Versions
         let versions = [];
-        // IF this is a new (or existing) object in COMS DB (from step 1)
-        // OR doing 'full' sync AND object wasn't deleted
```
Comment (TimCsaky) [inline app/src/services/sync.js:45]: some kind of comments here explaining the logic are useful. we could add them later.
I think the current general behaviour is that **without** full mode, we only sync versions and meta/tags for **new** objects

**Review by TimCsaky — COMMENTED** [review]
some good optimizations here

**Pull request merged**